cmd/create/createCluster.go

```diff
@@ -0,0 +1,419 @@
+/*Package create ...
+
+Copyright © 2020 The k3d Author(s)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+package create
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	cliutil "github.com/rancher/k3d/cmd/util"
+	"github.com/rancher/k3d/pkg/cluster"
+	k3dCluster "github.com/rancher/k3d/pkg/cluster"
+	"github.com/rancher/k3d/pkg/runtimes"
+	k3d "github.com/rancher/k3d/pkg/types"
+	"github.com/rancher/k3d/version"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// NewCmdCreateCluster returns a new cobra command
+func NewCmdCreateCluster() *cobra.Command {
+
+	createClusterOpts := &k3d.CreateClusterOpts{}
+
+	// create new command
+	cmd := &cobra.Command{
+		Use:   "cluster NAME",
+		Short: "Create a new k3s cluster in docker",
+		Long:  `Create a new k3s cluster with containerized nodes (k3s in docker).`,
+		Args:  cobra.RangeArgs(0, 1), // exactly one cluster name can be set (default: k3d.DefaultClusterName)
+		Run: func(cmd *cobra.Command, args []string) {
+			cluster := parseCreateClusterCmd(cmd, args, createClusterOpts)
+			if err := k3dCluster.CreateCluster(cluster, runtimes.SelectedRuntime); err != nil {
+				log.Errorln(err)
+				log.Errorln("Failed to create cluster >>> Rolling Back")
+				if err := k3dCluster.DeleteCluster(cluster, runtimes.SelectedRuntime); err != nil {
```

Comment:
If you use the same cluster name twice.  The second time it fails, but then deletes the first cluster on the second time.

Reply:
True! Now checking if a cluster with the chosen name exists before doing anything at all :+1: